compiler/rustc_typeck/src/outlives/explicit.rs

```diff
@@ -58,6 +58,7 @@ impl<'tcx> ExplicitPredicatesMap<'tcx> {
                     | ty::PredicateAtom::Subtype(..)
                     | ty::PredicateAtom::ConstEvaluatable(..)
                     | ty::PredicateAtom::ConstEquate(..) => (),
+                    ty::PredicateAtom::TypeFromEnv(..) => unimplemented!(),
```

Comment:
What would you think about changing this list (which is also a duplicate) to a `_` pattern?